src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -136,6 +143,8 @@ public static CodeWriter.CodeWriterScope WriteMethodDeclaration(this CodeWriter
                 .Append($"{methodBase.Name}(")
                 .AppendRawIf("this ", methodBase.Modifiers.HasFlag(Extension));
 
+            var outerScope = writer.AmbientScope();
```

Comment:
When method is declared, we need two scopes: inner for the body and outer that will include parameters.

Reply:
Yes. Scopes require some rework, but it would be a big change in the code that uses it, so it will be a separate work item.